haystack/preview/dataclasses/answer.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass
+from haystack.preview.dataclasses.document import Document
+
+
+@dataclass(frozen=True)
+class Answer:
+    data: Any
+    question: str
+    metadata: Dict[str, Any]
+
+
+@dataclass(frozen=True)
+class ExtractiveAnswer(Answer):
+    data: Optional[str]
```

Comment:
On a second thought I'm wondering if this typing is correct. Now we need `str`, but just as in the case of Document, the typing here is non-trivial because soon we'll need it to support tables, images, binary blobs like images or audio.

I'd say that for now it's ok to have `str`, but we should keep in mind to update these classes as soon as we decide how to address this problem in the Document's case.